### PR TITLE
Add logic for recover password component

### LINKED ISSUE #13: Реализовать сброс пароля пользователя

### PR DESCRIPTION
**Pull request by ilyamore88**
Closes #13

### CODE REVIEW

File: src/api/user/index.js [hunk 1]
```diff
@@ -35,6 +36,16 @@ export async function signUp(email) {
...
 }
 
+/**
+ * Recover password by email
+ *
+ * @param {string} email - user Email
+ * @return {Promise<boolean>} Response status
+ */
+export async function recoverPassword(email) {
+  return (await api.call(MUTATION_RECOVER_PASSWORD, { email })).resetPassword;
```
Comment (neSpecc) [inline src/api/user/index.js]: what is `resetPassword`?

File: src/components/auth/RecoverPassword.vue [hunk 2]
```diff
@@ -48,6 +62,23 @@ export default class RecoverPassword extends Vue {
...
   }
+
+  /**
+   * Method recover user's password by email from form
+   */
+  private async recoverPassword(): Promise<void> {
+    const email = this.fields[0].value;
+
+    try {
+      await userApi.recoverPassword(email);
```
Comment (neSpecc) [inline src/components/auth/RecoverPassword.vue]: better call api through the vuex action

File: src/components/auth/RecoverPassword.vue [hunk 3]
```diff
@@ -48,6 +62,23 @@ export default class RecoverPassword extends Vue {
     ];
     this.submitText = this.$t('authPages.recoverPassword');
   }
+
+  /**
+   * Method recover user's password by email from form
+   */
+  private async recoverPassword(): Promise<void> {
+    const email = this.fields[0].value;
```
Comment (neSpecc) [inline src/components/auth/RecoverPassword.vue]: better search by name

File: src/components/auth/RecoverPassword.vue [hunk 4]
```diff
@@ -48,6 +62,23 @@ export default class RecoverPassword extends Vue {
...
+
+    try {
+      await userApi.recoverPassword(email);
+      this.$router.push('/login');
```
Comment (neSpecc) [inline src/components/auth/RecoverPassword.vue]: how I will know the new password?

Comment (ilyamore88) [reply src/components/auth/RecoverPassword.vue]: The new password will be sent to the email

File: src/api/user/index.js (hunk 1 above)
Comment (ilyamore88) [reply src/api/user/index.js]: This is a mutation in graphql api. To get only `true` response you need to write this. Else you get response like object
```
{
  resetPassword: true
}
```

File: src/components/auth/RecoverPassword.vue [hunk 5]
```diff
@@ -48,6 +62,24 @@ export default class RecoverPassword extends Vue {
...
+  /**
+   * Method recover user's password by email from form
+   */
+  private async recoverPassword(): Promise<void> {
+    const emailField = this.fields.find(field => field.name === 'email');
+    const email = emailField ? emailField.value : '';
+
+    try {
+      await this.$store.dispatch(RECOVER_PASSWORD, { email });
+      this.$router.push('/login');
```
Comment (neSpecc) [inline src/components/auth/RecoverPassword.vue]: let's pass a get parameter and show success-message on login screen
image

**Pull request merged**